### PR TITLE
Add TLC model-pair validation and integration coverage

### PR DESCRIPTION
**Pull request by younes-io**
- expose TLC runner overrides from activation so tests can stub TLC processes and reset cleanly
- validate model/spec pairing by reading EXTENDS lines (with inline comment stripping) before launching TLC and warn on mismatches
- adjust TLC stdout parsing edge case and ensure pretest build pipeline compiles extension artifacts
- add end-to-end “model check” integration tests plus fixture workspaces that exercise custom runs, multi-root setups, and library dependencies

### CODE REVIEW

File: src/commands/checkModel.ts [hunk 1]
```diff
@@ -316,6 +351,71 @@ async function checkModelExists(cfgPath: string, warn = true): Promise<boolean>
...
+                break;
+            }
+        }
+    } catch (err) {
+        console.warn(`Unable to inspect model file ${modelPath}: ${err}`);
+    }
+    return undefined;
+}
+
+async function validateModelPairing(
```
Comment (lemmy) [inline src/commands/checkModel.ts]: Below is a perfectly valid model that uses `INSTANCE` instead of `EXTENDS`.

```tla
----- MODULE Counter ------

CONSTANT S
VARIABLE v

Spec == v = 0 /\ [][v' \in S]_v
=====
```

```tla
----------------------------- MODULE MCCounter ------------------------------ 
VARIABLE v

INSTANCE Counter WITH S <- {1,2,3}
=============================================================================
```

```
SPECIFICATION Spec
```

Further thought and experimentation are required regarding nested modules.

File: src/commands/checkModel.ts [hunk 2]
```diff
@@ -316,6 +351,71 @@ async function checkModelExists(cfgPath: string, warn = true): Promise<boolean>
     return cfgExists;
 }
 
+async function readExtendsModules(modelPath: string): Promise<string[] | undefined> {
```
Comment (lemmy) [inline src/commands/checkModel.ts]: Using SANY instead of regular expressions would be simpler and more robust.

Comment (younes-io) [reply src/commands/checkModel.ts]: dealt with in the last commit

File: src/commands/modelReferences.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+import { runXMLExporter } from '../tla2tools';
+
+export interface ModuleReferences {
+    extends?: Set<string>;
+    instances?: Set<string>;
+}
+
+/**
+    Extracts module references from a model file.
+    - EXTENDS clauses are read from the source text to avoid false positives when SANY
```
Comment (lemmy) [inline src/commands/modelReferences.ts]: I don't see a reason why SANY's XML exporter cannot include this information.  Let's get this feature added instead of relying on parsing the file.
